tests/classes/apiRest/apiRest_test.php

```diff
@@ -151,6 +151,87 @@ public function test_get_credential() {
         $this->assertEquals($result, $resdata);
     }
 
+    /**
+     * Tests if `POST /v1/issuer/groups/search` is properly called.
+     */
+    public function test_search_groups() {
```

Comment:
Is there a better way to split these tests out? The comments are clear, but separate functions within some sort of broader block structure would seem better, but I'm not sure with PHP what the norms are.

Reply:
@alanheppenstall - I'm not sure about the PHPUnit practice too, but looking at unit tests of other Moodle plugins, it seems like there are no broader block structures in PHPUnit like `context` in RSpec.

I could find a couple of examples that separate different scenarios of the same function to different testing functions and name them like `test_method_when_scenario1` or `test_method_with_scenario1`.

For example:
- [moodle-mod_zoom](https://github.com/ncstate-delta/moodle-mod_zoom/blob/master/tests/mod_zoom_invitation_test.php)
- [moodle-mod_attendance](https://github.com/danmarsden/moodle-mod_attendance/blob/MOODLE_311_STABLE/tests/externallib_test.php)